docker-compose.yml

```diff
@@ -0,0 +1,87 @@
+version: "3"
+
+services:
+  database:
+    container_name: DarkflameDatabase
+    image: mariadb:10.6
+    environment:
+      - MARIADB_USER=${MARIADB_USER:-darkflame}
+      - MARIADB_PASSWORD=${MARIADB_PASSWORD:-darkflame}
+      - MARIADB_ROOT_PASSWORD=${MARIADB_ROOT_PASSWORD:-darkflame}
+      - MARIADB_DATABASE=${MARIADB_DATABASE:-darkflame}
+    volumes:
+      - database:/var/lib/mysql
+      - ./migrations/dlu:/docker-entrypoint-initdb.d
+    networks:
+      - darkflame
+
+  darkflame:
+    container_name: DarkFlameServer
+    networks:
+      - darkflame
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+      args:
+        - BUILD_THREADS=${BUILD_THREADS:-1}
+        - BUILD_VERSION=${BUILD_VERSION:-171022}
+    environment:
+      - DATABASE=${MARIADB_DATABASE:-darkflame}
+      - DATABASE_HOST=database
+      - DATABASE_USER=${MARIADB_USER:-darkflame}
+      - DATABASE_PASSWORD=${MARIADB_PASSWORD:-darkflame}
+    volumes:
+      - ${CLIENT_PATH:?err}:/client
+    depends_on:
+      - database
+    ports:
+      - "1001:1001/udp"
+      - "2000:2000/udp"
+      - "2005:2005/udp"
+      - "3000-3300:3000-3300/udp"
```

Comment:
FYI, DLU just allocates ports with increasing port numbers every time a new zone is started - this could (theoretically) go all the way up to 65536

Reply:
Yeah, but we shouldn't open all this ports. If this really happens, the user could increase the port range by itself.

Reply:
I agree. Limiting to 300 as the default seems sane to me. I was also thinking today that using ports for this is probably not a great idea. It's probably better to have some sort of proxy (nginx?) servicing endpoints to the client instead of all these ports being open on the host.

Reply:
+1 for the proxy. But this would have to be supported by both the server and the client, with the latter probably being harder to change. For example, if the client just asks for a port when changing worlds, this would be a problem.